### ISSUE TITLE
Data generation

### ISSUE THREAD

**Issue by kayacan459** [alan-turing-institute/turing-deinterleaving-challenge#8]
Dear Dr. Groves and Dr. Nockles,

I hope this message finds you well.

My name is Mehmet Kayacan, and I am currently working on a radar signal processing project focused on pulse deinterleaving and synthetic PDW dataset generation. I recently came across the Turing Synthetic Radar Dataset (TSRD) and the Turing Deinterleaving Challenge, and I found the work extremely relevant and inspiring for my research.

I am particularly interested in understanding the simulation and data generation pipeline used to create the TSRD. I have reviewed the public GitHub repository and Hugging Face dataset page, but I could not find information regarding whether the simulator or dataset generation code is publicly available.

If possible, I would greatly appreciate any clarification on:

Whether the simulation/data generation framework is available publicly,
Or if there are any technical reports, documentation, or publications describing the generation methodology in more detail.
Thank you very much for your time and for making this valuable dataset available to the research community. I look forward to your response.

Best regards,
Mehmet KAYACAN

**Comment by lltiez** [comment]
Hi @kayacan459 , the code is not publicly available, but you can find the paper with all details here https://arxiv.org/pdf/2602.03856. If you found the TSRD useful for your own research, please consider citing.

> Gunn, E., Hosford, A., Jones, R., Zeitler, L., Groves, I. and Nockles, V., 2026. The Turing Synthetic Radar Dataset: A dataset for pulse deinterleaving. arXiv preprint arXiv:2602.03856.

**Comment by kayacan459** [comment]
Hi,

Thank you for your response and for updating the documentation regarding
the PDW definition. We reviewed the updated information and also looked
again at the paper, especially the reported AMI, ARI, V-measure,
homogeneity, and completeness scores. These results are very helpful for
understanding the dataset and the expected behavior of the deinterleaving
pipeline.

We also ran some additional experiments on the dataset. When we applied
HDBSCAN directly on raw PDW values without normalization, the performance
was relatively low, which was expected due to the different scales of ToA,
RF, PW, AoA, and amplitude. However, after applying feature normalization
and using the normalized PDW features with HDBSCAN, we obtained noticeably
higher scores. In our experiments, the normalized feature + HDBSCAN
baseline achieved approximately AMI ≈ 0.78, ARI ≈ 0.60, and V-measure ≈
0.78 on the filtered evaluation setup.

Because of this, we wanted to ask whether you also tested a normalized PDW
+ HDBSCAN baseline, and if so, how you interpret its performance. In some
cases, the normalized PDW features seem to separate the emitters quite well
even without a learning-based model. Do you think this behavior is expected
due to the simulator configuration, or could it indicate that some emitter
parameters in the synthetic data are more separable than they would be in a
realistic radar environment?

More generally, we would appreciate your thoughts on how to interpret the
realism of the dataset from a deinterleaving perspective. The paper results
suggest that the dataset is challenging and meaningful, especially in scan
scenarios and with overlapping emitters, but our normalized HDBSCAN results
made us wonder how close the parameter distributions are to realistic dense
electronic warfare environments.

Thank you again for your help and for making the dataset available.

Best regards,
Mehmet KAYACAN

lltiez ***@***.***>, 1 Haz 2026 Pzt, 19:03 tarihinde şunu
yazdı:

**Comment by lltiez** [comment]
Hi @kayacan459 ,

yes, this is expected. HDBscan works on densities and needs to measure distances first. If not normalised properly, the distances along one axis can easily dominate other PDW properties. I'd say the most important thing you should spend time on is how these distances (and therefore the densities) are calculated and what assumptions your distance calculations include. If you used the scikit-learn package, you'll notice that the default metric is Euclidean. The assumptions made by Euclidean distances on higher-dimensional PDWs are likely not a good metric for deinterleaving. That leaves you with two options: either project PDWs into a space where Euclidean distance can be applied for deinterleaving, or you design your own distance metric that approximates the manifold which the PDWs span. 

"Simply" playing around with the clustering algorithm will bring you a long way. I say "simply" in quotes, because, honestly, this is the actual work you should be doing. We report only the un-normalised results in the paper because that's the most naive benchmark.

I'll keep the issue closed, but again, feel free to re-open if you think that this hasn't been sufficiently addressed. 

PS: we have traditionally focused on V-measures, but different clustering scores capture different properties. Don't under-estimate the use of the pairwise metrics, as they give you a better understanding for an average worst-case performance (i.e. emitters that produce only a handful of pulses and how well you can separate those).

**Comment by ecatixtsc** [comment]
Hi both,

Interesting conversation. I have noticed the same phenomenon with HDBSCAN.
Another note, a 1000-pulse window has a highly skewed emitter-size distribution (a few emitters carry most pulses), so any pulse-weighted score reflects the bulk and is nearly blind to per-emitter errors.  Such as, I have used a composite score where v-measure is only 25% of the metric and **MCC and F1** have a share into the composite as well.
In scan/training I found that in average, one emitter has 63% of pulses so each frame of reference would be characterized by class imbalance.
 (V-measure, accuracy) reflect the majority structure and are insensitive to per-emitter errors; also V-measure is not chance-corrected and is inflated by degenerate over-segmentation.
Then I have tried a self-supervised model. Good scores, however, when I have introduced ablations, I have noticed AoA dominates the model's decisions almost entirely!
When AOA is removed / flattened to a constant value, the clustering collapses and the V-measure drops sharply, whereas applying the same treatment to RF or PW barely moves the scores. The implication is that the strong scores are, in part, an artefact of the dataset, in which distinct bearings happen to separate most emitters; the model has learned a single-feature rather than a multi-parameter representation.
Then I have separated all cases where the model was doing well and evaluated only the difficult ones and this is when the real work started. 89% of the model's failures are for the co-located emitters with SwitchDwell, RF agility and PW switching.

**Comment by lltiez** [comment]
Hi @ecatixtsc ,

that's great, love that community discussion. 

Re the importance of the AoA: I'm not surprised by this, any geolocation information should immensely help deinterleaving (for example the same emitter at two different locations). Thanks for pointing this out! I think that, for any real-world scenario, a model should use this information if available, so I wouldn't say that this is an artifact. I reckon a different question is how robust is the model to removing certain bits of information. It could be worth it to introduce sub-challenges (e.g. model performance without spatial information in the input).

Re the imbalance: imbalance is deliberate (see the paper). Thanks for highlighting the importance of the other metrics for your work. This is why we have them in there! As mentioned in the comment before, we have traditionally focused on v-measures as metric, but it is unlikely to have one model that rules them all, and rather, there are different methods that are particularly useful for different metrics. However, as pointed out simply clustering (including on normalised PDWs) won't give you amazing v-scores either (you v-measures should be in the 0.9). Note that even chance correction doesn't correct for the class imbalance (ARI and AMI are chance corrected, and v-measures tend to be very close to AMIs in my experience).

Re
> 89% of the model's failures are for the co-located emitters with SwitchDwell, RF agility and PW switching.

That's a great shout. Future releases could focus on these emitter behaviours. In the mean time, I think filtering those out to challenge model development is a good approach

**Comment by ecatixtsc** [comment]
We have a model that has been trained on scan/train for 25 epochs, below are the evaluation results.

For archive/test we skipped the nan and inf data, as you can see, teh MCC and F1 really drops.
We trained and evaluated on 1000 pulse windows.
The model's weaknesses are: co-located emitters with SwitchDwell, frequency hopping, PW switching and true emitters > 10.
The problem is not currently solved. :)

<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:w="urn:schemas-microsoft-com:office:word"
xmlns:m="http://schemas.microsoft.com/office/2004/12/omml"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Word.Document>
<meta name=Generator content="Microsoft Word 15">
<meta name=Originator content="Microsoft Word 15">
<link rel=File-List
href="file:///C:/Users/ecati/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">

<link rel=themeData
href="file:///C:/Users/ecati/AppData/Local/Temp/msohtmlclip1/01/clip_themedata.thmx">
<link rel=colorSchemeMapping
href="file:///C:/Users/ecati/AppData/Local/Temp/msohtmlclip1/01/clip_colorschememapping.xml">





</head>

<body lang=EN-GB style='tab-interval:.5in;word-wrap:break-word'>



Metric | scan/test | archive/test | stare/test
-- | -- | -- | --
V-measure | 0.9899 | 0.9781 | 0.9901
MCC | 0.8708 | 0.4855 | 0.7375
F1 | 0.8585 | 0.4138 | 0.7033
ARI | 0.9889 | 0.9613 | 0.9933
AMI | 0.9897 | 0.9763 | 0.9896
Homogeneity | 0.9908 | 0.9683 | 0.9954
Completeness | 0.9912 | 0.9897 | 0.9853
Coverage | 26,115/26,115 (100%) | 4,332/33,771 (12.8%) | 30,000/365,943 (8.2%)




</body>

</html>

**Comment by lltiez** [comment]
Re-opening this now, cos I think that's very interesting discussion.

@ecatixtsc These are great results. You got very high MCC and F1 scores as well, that's impressive. Would you be interested in submitting your results so we can add them to our leader board? We'd be generating a completely new test dataset without labels, and you can return to us only the labels. We'd then compare your labels with the ground truth. You can opt for an anonymous submission as well, where would only publish your results without name or affiliation.  If that sounds like something you'd be interested in, please get in touch using the contact details on this repo.

The archive version is deprecated (we found couple of bugs which is why we updated and re-generated everything), and we only keep that for legacy purposes. I hope you didn't find any NaNs/infs in the new version!? If so, please report a separate issue and I will look into this.

**Comment by ecatixtsc** [comment]
Hi @lltiez . Thank you for the updates, really looking forward to a new dataset since these metrics point out to clear overfitting.
It is ok to add the name and the organization in the published results.

**Comment by lltiez** [comment]
@ecatixtsc what do you mean with overfitting?

a) If it's that your model can't generalise (which doesn't look like it according to the numbers), then this is a problem independent of the dataset, and you need to change your training. 
b) If you think the dataset is too easy, then this has nothing to do with overfitting. If that's the case, please consider highlighting what you'd expect in a new dataset, so that we can address them as feature requests. Please feel free to open a new github issue.
c) If you think the data is unrealistic, then this might be possibly bug. We intend to simulate (within limits) realistic emitter behaviours. If that's the case, I'd encourage you to report them as a separte bug (containing problem description, expected behaviour, and actual behaviour---if possible also add what environment you ran that on, such as python, library, and data version, as well as a short code snippet how to reproduce the issue)

**Comment by ecatixtsc** [comment]
The test dataset has a distribution that is very similar to that of the training dataset so it is hard to know if the model generalizes,

**Comment by lltiez** [comment]
Hi @ecatixtsc, just to for me to understand you correctly: is your feature request an additional out-of-distribution test set?

**Comment by ecatixtsc** [comment]
@lltiez Apologies, I do not want to make work for you.
The model was trained on scan/train. 
scan/test scores high mainly because it's in-distribution 
archive/test is out-of-distribution its much higher emitter density (~25 vs ~7 per 1,000-pulse window) is what collapses MCC/weakest-emitter F1 via minority over-merge, even though V stays high.
So archive data has out of distribution examples, it would be nice to have more of it :)

**Comment by lltiez** [comment]
@ecatixtsc OK got you now. That's a great shout, I think the best would be holding out certain emitter types completely, that'd be the most realistic scenario for unseen data points. I'll have a look at this and put this forward for our next release. Thanks for your detailed feedback!
